File: programmers/Level2/CPP/스킬트리.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(string skill, vector<string> skill_trees) {
    int answer = 0;
    
    for(int i = 0; i< skill_trees.size(); i++){
        int num = 0;
        bool b = 0;
        for(int j = 0; j < skill_trees[i].size(); j++){
            if(skill.find(skill_trees[i][j]) == string::npos){
                continue;
            }
            
            if(skill.find(skill_trees[i][j])!=num){
                b = 1;
            }
            num++;
        }
        
        if(b == 0) answer++;  
        
    }
    
    return answer;
}

File: programmers/Level2/JAVA/스킬트리.java
class Solution {
    public int solution(String skill, String[] skill_trees) {
        int answer = 0;
        for(int i = 0; i< skill_trees.length; i++){
        int num = 0;
        boolean b = false;
        for(int j = 0; j < skill_trees[i].length(); j++){
            if(skill.indexOf(skill_trees[i].charAt(j)) == -1){
                continue;
            }
            
            if(skill.indexOf(skill_trees[i].charAt(j))!=num){
                b = true;
            }
            num++;
        }
        
        if(b == false) answer++;  
        
    }
        return answer;
    }
}
